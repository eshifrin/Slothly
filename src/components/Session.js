import React, { Component } from 'react';
import { StyleSheet, View, Text, TouchableHighlight } from 'react-native';
import Separator from './Separator';
import { colors } from '../constants';
import RenderIf from '../utils/RenderIf';
import { formatDate } from '../utils/calcStats';
import { UpIcon, DownIcon } from '../utils/Icons';
import PropTypes from 'prop-types';

export default class Session extends Component {
  state = {
    showNote: false
  };

  render() {
    const { item } = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.rowContainer}>
          <Text style={styles.rowText}>{item.sessionLength}</Text>
          <Text style={styles.rowText}>{formatDate(item.when)}</Text>
          <TouchableHighlight
            onPress={() => this.setState({ showNote: !this.state.showNote })}
            underlayColor="transparent"
          >
            <Text>{this.state.showNote ? UpIcon : DownIcon}</Text>
          </TouchableHighlight>
        </View>

        <RenderIf condition={this.state.showNote}>
          <View style={styles.noteContainer}>
            <Separator />
            <Text style={styles.noteHeader}>SESSION NOTES</Text>
            <Text style={styles.note}>{item.note}</Text>
          </View>
        </RenderIf>
      </View>
    );
  }
}

Session.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string,
    sessionLength: PropTypes.oneOfType(PropTypes.number, PropTypes.string),
    when: PropTypes.instanceOf(Date),
    note: PropTypes.string
  })
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: colors.deepBlue
  },
  rowContainer: {
    flex: 1,
    justifyContent: 'space-around',
    flexDirection: 'row',
    alignItems: 'center',
    height: 75
  },
  rowText: {
    color: 'white',
    fontFamily: 'Futura',
    fontSize: 18
  },
  noteHeader: {
    fontWeight: '700',
    backgroundColor: colors.lightBLue,
    color: colors.deepBlue,
    fontFamily: 'Futura',
    fontSize: 13,
    flex: 1,
    height: 20,
    textAlign: 'center'
  },
  note: {
    backgroundColor: colors.lightBLue,
    color: colors.deepBlue,
    fontFamily: 'Futura',
    fontSize: 15,
    flex: 1
  },
  noteContainer: {
    backgroundColor: colors.lightBLue,

    flex: 1,
    flexDirection: 'column',
    paddingLeft: 15
  }
});
