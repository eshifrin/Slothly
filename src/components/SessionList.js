import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  FlatList,
  Text,
  ImageBackground,
  TouchableHighlight
} from 'react-native';
import Separator from './Separator';
import Session from './Session';
import RenderIf from '../utils/RenderIf';
import { SwipeListView } from 'react-native-swipe-list-view';
import PropTypes from 'prop-types';

export default class SessionList extends Component {
  static defaultProps = {
    sessions: []
  };

  render() {
    return (
      <SwipeListView
        disableRightSwipe
        useFlatList
        style={styles.container}
        data={this.props.sessions}
        renderItem={this._renderSession}
        keyExtractor={this._keyExtractor}
        renderHiddenItem={this._renderHiddenItem}
        rightOpenValue={-140}
        ItemSeparatorComponent={Separator}
      />
    );
  }

  _keyExtractor = (item, index) => `${item.id}`;
  _renderSession = ({ item }) => <Session item={item} />;
  _renderHiddenItem = ({ item, index }) => {
    return (
      <View style={styles.buttonContainer}>
        <TouchableHighlight
          style={[styles.button, styles.editButton]}
          onPress={() => this.props.editSession(index)}
        >
          <Text>EDIT</Text>
        </TouchableHighlight>
        <TouchableHighlight
          style={[styles.button, styles.deleteButton]}
          onPress={() => this.props.deleteSession(item.id)}
        >
          <Text>DELETE</Text>
        </TouchableHighlight>
      </View>
    );
  };
}

SessionList.propTypes = {
  sessions: PropTypes.array.isRequired,
  editSession: PropTypes.func.isRequired,
  deleteSession: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column'
  },
  buttonContainer: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'flex-end'
  },
  button: {
    flexBasis: 70,
    justifyContent: 'center',
    alignItems: 'center'
  },
  editButton: {
    backgroundColor: '#edf7ee'
  },
  deleteButton: {
    backgroundColor: '#efdbc2'
  }
});
