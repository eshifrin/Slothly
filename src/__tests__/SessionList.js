import 'react-native';
import React from 'react';
import SessionList from '../components/SessionList';
import Session from '../components/Session';
import { shallow } from 'enzyme';

describe('SessionList', () => {
  let wrapper;
  const mockSessionList = [
    {
      sessionLength: '5',
      when: new Date(2018, 0, 1),
      note: 'mock note'
    }
  ];

  const mockEditSession = jest.fn();
  const mockDeleteSession = jest.fn();

  beforeEach(done => {
    wrapper = shallow(
      <SessionList
        sessions={mockSessionList}
        editSession={mockEditSession}
        deleteSession={mockDeleteSession}
      />
    );
    setTimeout(() => done(), 0);
  });

  it('renders', () => {
    wrapper;
    expect(wrapper).toMatchSnapshot();
  });
});
