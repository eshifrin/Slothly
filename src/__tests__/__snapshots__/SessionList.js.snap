// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionList renders 1`] = `
<SwipeListView
  ItemSeparatorComponent={[Function]}
  closeOnRowBeginSwipe={false}
  closeOnRowPress={true}
  closeOnScroll={true}
  data={
    Array [
      Object {
        "note": "mock note",
        "sessionLength": "5",
        "when": 2018-01-01T08:00:00.000Z,
      },
    ]
  }
  directionalDistanceChangeThreshold={2}
  disableLeftSwipe={false}
  disableRightSwipe={true}
  keyExtractor={[Function]}
  leftOpenValue={0}
  previewFirstRow={false}
  recalculateHiddenLayout={false}
  renderHiddenItem={[Function]}
  renderItem={[Function]}
  rightOpenValue={-140}
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
  swipeToClosePercent={50}
  swipeToOpenPercent={50}
  swipeToOpenVelocityContribution={0}
  useFlatList={true}
/>
`;
